scm/githubv4.go

```diff
@@ -0,0 +1,83 @@
+package scm
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v35/github"
+	"github.com/shurcooL/githubv4"
+	"go.uber.org/zap"
+	"golang.org/x/oauth2"
+)
+
+type GithubV4SCM struct {
+	GithubSCM
+	clientV4     *githubv4.Client
+	bypassClient *github.Client
+}
+
+func NewGithubV4SCMClient(logger *zap.SugaredLogger, token string) *GithubV4SCM {
+	src := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	httpClient := oauth2.NewClient(context.Background(), src)
+	client := github.NewClient(httpClient)
+	return &GithubV4SCM{
+		GithubSCM: GithubSCM{
+			logger: logger,
+			client: client,
+			token:  token,
+		},
+		clientV4:     githubv4.NewClient(httpClient),
+		bypassClient: client,
+	}
+}
+
+func (s *GithubV4SCM) DeleteIssue(ctx context.Context, opt *RepositoryOptions, issueNumber int) error {
```

Comment:
With the inclusion of a new `DeleteIssue()` function, we can in theory delete issues, instead of marking them as deleted and closing them, whenever a teacher deletes a task.